### PR TITLE
Fix speculative decoding test oracle

### LINKED ISSUE #353: [BUG] SpeculativeDecodingTests fail sometimes
See #326 #315 and #323 plus recent failures in #344 

https://github.com/ml-explore/mlx-swift-lm/actions/runs/27645347325/job/81756903393?pr=344

```
◇ Test case passing 2 arguments numDraftTokens → 2, withLogitProcessor → false to "Speculative decoding matches default token generation" started.
✘ Test "Speculative decoding matches default token generation" recorded an issue with 2 arguments numDraftTokens → 2, withLogitProcessor → false at SpeculativeDecodingTests.swift:102:9: Expectation failed: (normalTokens → [39, 96, 93, 39, 39, 39, 39, 39, 39, 39, 37, 24, 24, 24, 17, 1, 77, 61, 59, 4, 1, 24, 61, 37, 99, 61, 37, 24, 17, 37, 24, 17]) == (speculativeTokens → [39, 37, 72, 37, 72, 37, 72, 96, 72, 96, 72, 59, 1, 65, 73, 1, 99, 24, 61, 59, 39, 41, 41, 24, 61, 59, 13, 64, 56, 38, 24, 61])
± inserted [93, 39, 39, 39, 39, 39, 39, 37, 24, 24, 17, 1, 77, 4, 1, 61, 37, 99, 37, 24, 17, 37, 24, 17], removed [37, 72, 37, 72, 37, 72, 96, 72, 72, 59, 1, 65, 73, 1, 99, 24, 61, 59, 41, 41, 13, 64, 56, 38]
```

I can get this to happen reliably by running the test in a loop -- it takes 2 or 3 iterations on my machine.

- `MLX_ENABLE_TF32=0` does not affect it (it shouldn't in #326 we switched to float16 to avoid this issue)
- curiously fixing the random seed also does not fix it (it doesn't prevent it or make it happen reliably)

### PR DESCRIPTION
**Pull request by aleroot**
Use a deterministic high-margin transition model for exact speculative-vs-greedy equality coverage. Keep Gemma3 in the suite as a smoke test because real MLX batched verification and token-by-token decoding can choose different argmaxes when logits are close.

## Proposed changes

Fix https://github.com/ml-explore/mlx-swift-lm/issues/353 

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the necessary documentation (if needed)

### CODE REVIEW

**Review by davidkoski — APPROVED** [review]
This makes sense to me -- thank you for thinking through this!
